scripts/release.sh

```diff
@@ -18,6 +19,22 @@ get_tag() {
   echo -n $vers
 }
 
+help() {
+  cat <<EOF
+Usage: $(basename $0) [OPTIONS]
+
+Options:
+  -d, --dry-run                 Output actions that would be taken, but don't run them.
+  -h, --help                    Display this text.
+  --private-registry <address>  Push the built images to the provided registry.
+  --skip-build                  Don't perform nix-build.
+  --skip-publish-to-dockerhub   Don't publish to Dockerhub.
+
+Examples:
+  $(basename $0) --private-registry 127.0.0.1:5000
+EOF
+}
```

Comment:
thanks for contributing usage message and arg parser - nice improvement! I would change the parameters to make them even more useful in general and to avoid code duplication.

`--private-registry` can become just `--registry` (if none is specified then dockerhub registry is assumed). Then you can leave the code as it is and just add registry prefix to built images when pushing to a particular registry. In docker speech, the target registry is nothing more than a prefix in the image name so we can keep that idea here.

`--skip-publish-to-dockerhub` can then be removed entirely or you could rename it to `--skip-publish` to skip the entire step of publishing images. But it is not needed now so perhaps better to remove it.

We can introduce a new `--tag` option to explicitly set a tag of built images and override the default logic when the tag is set based on the source branch. For ci that can be set to `ci`. For other use cases to something else.

Reply:
I tried altering the layout of the arguments as you specified, but I think it makes the code (and behaviour) more confusing than just having arguments to skip individual stages. Eg It's clear if you're just running locally you don't want to go near dockerhub so just run with --skip-push-to-dockerhub.

Reply:
Correct me if I'm wrong but for the purpose of producing CI images the way how it is used would change from:
```
release.sh --private-registry=x.y.z.t --skip-publish-to-dockerhub
```
to
```
release.sh --registry=x.y.z.t --tag=ci
```
I don't think that the later is more confusing in any way, though opinions on that might differ. Also with the original args there are combinations that don't make sense and there are combinations that would be valid use cases but are not possible. Examples of each follow.

This is certainly not what you want to do if you  publish images to two registries (with the new arguments that is impossible to do):
```
release.sh --private-registry=x.y.z.t
```

I want to build the images with the tag "watcher" and publish them to dockerhub registry so that someone else can test them too:
```
release.sh --tag=watcher
```

So it is basically turning the tool from CI specific tool to a swiss-knife. Though if you don't agree, we can leave that for later. I don't want to delay the PR if this is the last thing holding it.